otx/algorithms/classification/tasks/train.py

```diff
@@ -0,0 +1,191 @@
+
+from .inference import ClassificationInferenceTask
+
+logger = get_logger()
+
+TASK_CONFIG = ClassificationConfig
+
+
+class ClassificationTrainTask(ClassificationInferenceTask):
+    """Train Task Implementation of OTX Classification."""
+
+    def save_model(self, output_model: ModelEntity):
+        """Save best model weights in ClassificationTrainTask."""
+        logger.info("called save_model")
+        buffer = io.BytesIO()
+        hyperparams_str = ids_to_strings(cfg_helper.convert(self._hyperparams, dict, enum_to_str=True))
+        labels = {label.name: label.color.rgb_tuple for label in self._labels}
+        model_ckpt = torch.load(self._model_ckpt)
+        modelinfo = {
+            "model": model_ckpt["state_dict"],
+            "config": hyperparams_str,
+            "labels": labels,
+            "VERSION": 1,
+        }
+
+        torch.save(modelinfo, buffer)
+        output_model.set_data("weights.pth", buffer.getvalue())
+        output_model.set_data(
+            "label_schema.json",
+            label_schema_to_bytes(self._task_environment.label_schema),
+        )
+        output_model.precision = self._precision
+
+    def cancel_training(self):
+        """Cancel training function in ClassificationTrainTask.
+
+        Sends a cancel training signal to gracefully stop the optimizer.
+        The signal consists of creating a '.stop_training' file in the current work_dir.
+        The runner checks for this file periodically.
+        The stopping mechanism allows stopping after each iteration, but validation will still be carried out.
+        Stopping will therefore take some time.
+        """
+        self._should_stop = True
+        logger.info("Cancel training requested.")
+        if self.cancel_interface is not None:
+            self.cancel_interface.cancel()
+        else:
+            logger.info("but training was not started yet. reserved it to cancel")
+            self.reserved_cancel = True
+
+    def train(
+        self,
+        dataset: DatasetEntity,
+        output_model: ModelEntity,
+        train_parameters: Optional[TrainParameters] = None,
+    ):
+        """Train function in ClassificationTrainTask."""
+        logger.info("train()")
+        # Check for stop signal between pre-eval and training.
+        # If training is cancelled at this point,
+        if self._should_stop:
+            logger.info("Training cancelled.")
+            self._should_stop = False
+            self._is_training = False
+            return
+
+        # Set OTX LoggerHook & Time Monitor
+        update_progress_callback = train_default_progress_callback
+        if train_parameters is not None:
+            update_progress_callback = train_parameters.update_progress  # type: ignore
+        self._time_monitor = TrainingProgressCallback(update_progress_callback)
+        self._learning_curves = defaultdict(OTXLoggerHook.Curve)
+
+        stage_module = "ClsTrainer"
```

Comment:
How about moving `ClsTrainer` to `otx/algorithms/classification` as `backbone` did? 
Although `ClsTrainer` is defined in `external/MPA`, each backbone (`efficinetnet.py, mobilenetv3, ...`) is registered in `otx/algorithms/classification. I think the two implemented parts are located in a similar scope.

Reply:
About that, hope we can discuss at development meeting for further refactoring.